controllers/stats/EditorialReportComponentHandler.inc.php

```diff
@@ -0,0 +1,239 @@
+		$editorialStatistics = [];
+		$percentageField = 1;
+		$averageField = 2;
+		foreach ([
+			'SUBMISSION_RECEIVED' => [__('manager.statistics.editorial.submissionsReceived')],
+			'SUBMISSION_ACCEPTED' => [__('manager.statistics.editorial.submissionsAccepted')],
+			'SUBMISSION_DECLINED_TOTAL' => [__('manager.statistics.editorial.submissionsDeclined')],
+			'SUBMISSION_DECLINED_INITIAL' => ['&emsp;' . __('manager.statistics.editorial.submissionsDeclined.deskReject')],
+			'SUBMISSION_DECLINED' => ['&emsp;' . __('manager.statistics.editorial.submissionsDeclined.postReview')],
+			'SUBMISSION_DECLINED_OTHER' => ['&emsp;' . __('manager.statistics.editorial.submissionsDeclined.other')],
+			'SUBMISSION_PUBLISHED' => [__('manager.statistics.editorial.submissionsPublished')],
+			'SUBMISSION_DAYS_TO_FIRST_DECIDE' => [__('manager.statistics.editorial.averageDaysToDecide'), $averageField],
+			'SUBMISSION_DAYS_TO_ACCEPT' => ['&emsp;' . __('manager.statistics.editorial.averageDaysToAccept'), $averageField],
+			'SUBMISSION_DAYS_TO_REJECT' => ['&emsp;' . __('manager.statistics.editorial.averageDaysToReject'), $averageField],
+			'SUBMISSION_ACCEPTANCE_RATE' => [__('manager.statistics.editorial.acceptanceRate'), $percentageField],
+			'SUBMISSION_REJECTION_RATE' => [__('manager.statistics.editorial.rejectionRate'), $percentageField],
+			'SUBMISSION_DECLINED_INITIAL_RATE' => ['&emsp;' . __('manager.statistics.editorial.deskRejectRate'), $percentageField],
+			'SUBMISSION_DECLINED_RATE' => ['&emsp;' . __('manager.statistics.editorial.postReviewRejectRate'), $percentageField],
+			'SUBMISSION_DECLINED_OTHER_RATE' => ['&emsp;' . __('manager.statistics.editorial.otherRejectRate'), $percentageField]
+		] as $field => list($name, $type)) {
+			$isPercentage = $type == $percentageField;
+			$isAverage = $type == $averageField || $isPercentage;
+			$editorialStatistics[] = [
+				'name' => $name,
+				'period' => round($rangedStatistics[$field], $isPercentage ? 2 : 0) . ($isPercentage ? '%' : ''),
+				'average' => round($statistics[($isAverage ? '' : 'AVG_') . $field], $isPercentage ? 2 : 0) . ($isPercentage ? '%' : ''),
+				'total' => round($statistics[$field], $isPercentage ? 2 : 0) . ($isPercentage ? '%' : '')
+			];
+		}
+		return $editorialStatistics;
+	}
+
+	/**
+	 * Given an array with statistics extracts the required user statistics to display in the component
+	 * @param $rangedStatistics array An array with a subset of the results from a call to PKPStatsService::getUserStatistics
+	 * @param $statistics array An array with the results from a call to PKPStatsService::getUserStatistics
+	 */
+	public static function extractUserStatistics($rangedStatistics, $statistics) {
+		$userStatistics = [];
+		foreach ([0 => 'manager.statistics.editorial.registeredUsers'] + Application::getRoleNames(true) as $id => $role) {
+			$userStatistics[] = [
+				'name' => __($role),
+				'period' => (int)$rangedStatistics[$id]['total'],
```

Comment:
Let's check with @asmecher about his experience with linters/formatters (e.g. https://cs.symfony.com), would be better to have such things done automatically.

Reply:
Currently we're only using PHP Scrutinizer, which isn't picky about formatting. I'm open to proposals for PHP linting/formatting, though don't have specific experience -- however, my worry is that it'll generate thousands of lines of busywork changes without much of a benefit. If there's something with a good cost/benefit, I'm all ears!

Reply:
I'll add a reminder to check the available options. As PHP has a built-in tokenizer since a long time ago, I guess there must be something good, open-source, that runs offline and auto-fixes in a reliably way =]

Reply:
I'd really like a formatter. I've found the different styles of code to be found in the 10+ years of code we're working with can be disorienting. But the best part is just that it takes these decisions out of our hands. A good code formatter should be able to be run as a pre-commit hook, which would mean all PRs would arrive pre-formatter. And it should be able to format automatically, which is to stay we shouldn't have to manually reformat our existing codebase. I'd consider these requirements for a formatter choice. It shouldn't require busywork once it's configured.